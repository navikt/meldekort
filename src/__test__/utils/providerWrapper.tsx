import * as React from 'react';
import { store } from '../../app/store/configureStore';
import { IntlProvider } from 'react-intl-redux';
import { Provider } from 'react-redux';

interface Props {
  children: React.ReactNode;
}

const ProviderWrapper: React.FunctionComponent<Props> = props => {
  return (
    <Provider store={store}>
      <IntlProvider locale="nb" defaultLocale="nb">
        {props.children}
      </IntlProvider>
    </Provider>
  );
};

export default ProviderWrapper;
