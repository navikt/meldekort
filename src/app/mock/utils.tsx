export function erLocalhost() {
  const host: string = window.location.host;
  return host.includes('localhost') || host.includes('127.0.0.1');
}

export function erMock(): boolean {
  let meldekort = 'meldekort';
  window[meldekort] = {
    MELDEKORTSESSIONSTORAGE_USERNAME: 'username',
    MELDEKORTSESSIONSTORAGE_PASSWORD: 'password',
  };

  return process.env.REACT_APP_MOCK_FULL === 'true';
}
