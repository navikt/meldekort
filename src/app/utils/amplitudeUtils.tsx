import amplitude from 'amplitude-js';
import Environment from './env';

let initialized = false;

function initAmplitude() {
  const { amplitudeKey, amplitudeUrl } = Environment();

  if (!amplitudeKey || !amplitudeUrl) {
    return;
  }

  const config = {
    apiEndpoint: amplitudeUrl,
    saveEvents: true,
    includeUtm: true,
    includeReferrer: true,
    trackingOptions: {
      city: false,
      ip_address: false,
    },
  };
  amplitude.getInstance().init(amplitudeKey, undefined, config);
  initialized = true;
}

initAmplitude();

function amplitudeLogger(name: string, values?: object) {
  if (!initialized) {
    console.log('Aktivitet uten initialisert amplitude:', name, values);
    return;
  }
  amplitude.getInstance().logEvent(name, values);
}

type AmplitudeAktivitetsData = {
  arbeidssoker: string;
  meldegruppe: string;
  innsendingstype: string;
};

export function loggAktivitet(
  aktivitet: string,
  data?: AmplitudeAktivitetsData
) {
  const eventData = { ...data, aktivitet: aktivitet };
  amplitudeLogger('meldekort.aktivitet', eventData);
}
