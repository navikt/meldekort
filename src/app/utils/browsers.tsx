// Edge
export const isEdge = window.navigator.userAgent.indexOf('Edge') !== -1;

// Internet Explorer 6-11
export const isIE =
  window.navigator.userAgent.indexOf('Trident') !== -1 && !isEdge;

// Firefox
export const isFireFox = window.navigator.userAgent.indexOf('Mozilla') !== -1;

// Safari
export const isSafari = window.navigator.userAgent.indexOf('Safari') !== -1;
