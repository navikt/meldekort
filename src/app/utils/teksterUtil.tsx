import { hentIntl } from './intlUtil';
import { Meldegruppe } from '../types/meldekort';

export enum TypeYtelse {
  'AAP' = '-AAP',
  'DAGPENGER' = '',
  'TILTAKSPENGER' = '-TP',
}

export const finnTypeYtelse = (meldegruppe: Meldegruppe): string => {
  // Da er det mulig Ã¥ sjekke alt her og returnere TypeYtelse, eller bedre Ã¥ si tilsvarende postfix
  return meldegruppe === Meldegruppe.ATTF
    ? TypeYtelse.AAP
    : TypeYtelse.DAGPENGER;
};

export const finnesIntlId = (tekstid: string): string => {
  if (hentIntl().formatMessage({ id: tekstid }) !== tekstid) {
    return tekstid;
  } else {
    return tekstid.slice(0, -4);
  }
};
