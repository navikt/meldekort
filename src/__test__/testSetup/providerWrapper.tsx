import * as React from 'react';
import { store } from '../../app/store/configureStore';
import { IntlProvider, IntlState, updateIntl } from 'react-intl-redux';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import teksterJsonNb from '../../app/mock/responses/tekster_nb.json';
import { Konstanter } from '../../app/utils/consts';
import { Meldekort } from "../../app/types/meldekort";
import { InnsendingState } from "../../app/types/innsending";

interface Props {
  children: React.ReactNode;
  initialState?: Meldekort | InnsendingState;
  initialStateName?: string;
}

let intl: IntlState;

const ProviderWrapper: React.FunctionComponent<Props> = props => {
  let testStore;
  if (props.initialState !== undefined) {
    const { initialState, initialStateName } = props;
    const mockState = {
      [initialStateName!]: initialState,
      intl: intl,
    };
    const mockStore = configureStore();
    testStore = mockStore(mockState);
  } else {
    testStore = store;
  }

  return (
    <Provider store={testStore}>
      <IntlProvider
        locale={Konstanter.defaultLocale}
        defaultLocale={Konstanter.defaultLocale}
      >
        {props.children}
      </IntlProvider>
    </Provider>
  );
};

export const setLocalesBeforeAll = () => {
  beforeAll(async () => {
    store.dispatch(
      // @ts-ignore
      updateIntl({
        locale: Konstanter.defaultLocale,
        messages: teksterJsonNb,
      })
    );
    intl = { locale: Konstanter.defaultLocale, messages: teksterJsonNb };
  });
};

export default ProviderWrapper;
