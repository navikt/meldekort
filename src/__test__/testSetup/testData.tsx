import { MenyPunkt } from '../../app/utils/menyConfig';
import SendMeldekort from '../../app/sider/sendMeldekort/sendMeldekort';
import TidligereMeldekort from '../../app/sider/tidligereMeldekort/tidligereMeldekort';
import {
  KortStatus,
  KortType,
  Meldegruppe,
  Meldekort,
  MeldekortDag,
  Meldeperiode,
} from '../../app/types/meldekort';
import {
  Feilmelding,
  InnsendingState,
  UtfyllingFeil,
} from '../../app/types/innsending';
import { hentSporsmalConfig } from '../../app/sider/innsending/1-sporsmalsside/sporsmal/sporsmalConfig';
import { hentUtfyltDagConfig } from '../../app/sider/innsending/2-utfyllingsside/utfylling/utfyltDagConfig';

export const menyPunkterTEST: MenyPunkt[] = [
  {
    exact: true,
    component: SendMeldekort,
    tittel: 'sendMeldekort',
    tekstid: 'naviger.send',
    urlparam: '/send-meldekort',
    disabled: false,
  },
  {
    component: TidligereMeldekort,
    tekstid: 'sekundarmeny.tidligere',
    tittel: 'tidligereMeldekort',
    urlparam: '/tidligere-meldekort',
    disabled: false,
  },
];

export const mkDagTEST: MeldekortDag = {
  dag: 1,
  arbeidetTimerSum: 7.5,
  syk: false,
  annetFravaer: true,
  kurs: false,
};

const dateString = '2019-01-01T10:00:00Z';

export const mkDetaljerTEST = {
  id: '',
  meldekortId: 0,
  meldeperiode: '',
  arkivnokkel: '',
  kortType: KortType.KORRIGERT_ELEKTRONISK,
  meldeDato: new Date(dateString),
  lestDato: new Date(dateString),
  sporsmal: {
    annetFravaer: false,
    arbeidet: false,
    arbeidssoker: false,
    syk: false,
    kurs: false,
    signatur: false,
    meldekortDager: [],
  },
  begrunnelse: '',
};

export const sporsmalOgSvarTest = [
  {
    kategori: 'arbeid',
    sporsmal: 'sporsmal.arbeid',
    forklaring: 'forklaring.sporsmal.arbeid',
    svar: true,
    formatertDato: '20.01.2019',
  },
  {
    kategori: 'aktivitetArbeid',
    sporsmal: 'sporsmal.aktivitetArbeid',
    forklaring: 'forklaring.sporsmal.aktivitetArbeid',
    svar: false,
    formatertDato: '20.01.2019',
  },
];

const meldeperiodeTEST: Meldeperiode = {
  til: new Date(dateString),
  fra: new Date(dateString),
  kanKortSendes: true,
  kortKanSendesFra: new Date(dateString),
  periodeKode: '201901',
};

export const mkTEST: Meldekort = {
  meldekortId: 1,
  kortType: KortType.ELEKTRONISK,
  meldeperiode: meldeperiodeTEST,
  meldegruppe: Meldegruppe.DAGP,
  kortStatus: KortStatus.OPPRE,
  bruttoBelop: 0,
  mottattDato: new Date(dateString),
  korrigerbart: true,
};

export const mkDagerTEST: MeldekortDag[] = [
  {
    dag: 1,
    arbeidetTimerSum: 7.5,
    syk: false,
    annetFravaer: true,
    kurs: false,
  },
];

const feilmelding: Feilmelding = {
  feil: true,
  feilmelding: 'Test',
};

export const utfyllingFeilTEST: UtfyllingFeil = {
  feilIArbeid: feilmelding,
  feilIKurs: feilmelding,
  feilISyk: feilmelding,
  feilIFerie: feilmelding,
  feilKombinasjonDagpSykArbeid: false,
  feilKombinasjonDagpFravaerArbeid: false,
  feilKombinasjonAAPFravaerSyk: false,
  feilKombinasjonAAPFravaerArbeid: false,
  feilIArbeidetTimerHeleHalve: true,
  feilIArbeidetTimer: true,
  feilIDager: ['mandag'],
};

export const innsendingTEST: InnsendingState = {
  meldekortId: 0,
  korrigertMeldekortId: 0,
  innsendingstype: null,
  begrunnelse: {
    valgtArsak: '',
    erFeil: false,
  },
  sporsmalsobjekter: hentSporsmalConfig(),
  utfylteDager: hentUtfyltDagConfig(),
  meldekortdetaljer: {
    id: '',
    meldekortId: 0,
    meldeperiode: '',
    arkivnokkel: '',
    kortType: KortType.KORRIGERT_ELEKTRONISK,
    meldeDato: new Date(dateString),
    lestDato: new Date(dateString),
    sporsmal: {
      annetFravaer: false,
      arbeidet: false,
      arbeidssoker: false,
      syk: false,
      kurs: false,
      signatur: false,
      meldekortDager: [],
    },
    begrunnelse: '',
  },
  meldekortdetaljerInnsending: undefined,
  valideringsResultat: undefined,
};
