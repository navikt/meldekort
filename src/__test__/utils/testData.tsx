import { MenyPunkt } from '../../app/utils/menyConfig';
import SendMeldekort from '../../app/sider/sendMeldekort/sendMeldekort';
import TidligereMeldekort from '../../app/sider/tidligereMeldekort/tidligereMeldekort';

export const menyPunkterTEST: MenyPunkt[] = [
  {
    exact: true,
    component: SendMeldekort,
    tittel: 'sendMeldekort',
    tekstid: 'naviger.send',
    urlparam: '/send-meldekort',
    disabled: false,
  },
  {
    component: TidligereMeldekort,
    tekstid: 'sekundarmeny.tidligere',
    tittel: 'tidligereMeldekort',
    urlparam: '/tidligere-meldekort',
    disabled: false,
  },
];
