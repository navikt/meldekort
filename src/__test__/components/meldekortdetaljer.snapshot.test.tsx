import * as React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ProviderWrapper from '../testSetup/providerWrapper';
import Meldekortdetaljer from '../../app/components/meldekortdetaljer/meldekortdetaljer';
import { mkDetaljerTEST, mkTEST } from '../testSetup/testData';
import { TypeYtelse } from '../../app/utils/teksterUtil';

Enzyme.configure({ adapter: new Adapter() });

it('Meldekortdetaljer', () => {
  const tree = shallow(
    <ProviderWrapper initialStateName={'aktivtMeldekort'} initialState={mkTEST}>
      <Meldekortdetaljer
        aktivtMeldekort={mkTEST}
        typeYtelse={TypeYtelse.AAP}
        meldekortdetaljer={mkDetaljerTEST}
      />
    </ProviderWrapper>
  );
  expect(tree).toMatchSnapshot();
});
