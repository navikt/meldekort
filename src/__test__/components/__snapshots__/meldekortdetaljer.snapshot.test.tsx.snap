// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meldekortdetaljer 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(IntlProvider)
    defaultLocale="nb"
    locale="nb"
  >
    <Connect(Meldekortdetaljer)
      aktivtMeldekort={
        Object {
          "bruttoBelop": 0,
          "korrigerbart": true,
          "kortStatus": "OPPRE",
          "kortType": "ELEKTRONISK",
          "meldegruppe": "DAGP",
          "meldekortId": 1,
          "meldeperiode": Object {
            "fra": 2019-01-01T10:00:00.000Z,
            "kanKortSendes": true,
            "kortKanSendesFra": 2019-01-01T10:00:00.000Z,
            "periodeKode": "201901",
            "til": 2019-01-01T10:00:00.000Z,
          },
          "mottattDato": 2019-01-01T10:00:00.000Z,
        }
      }
      meldekortdetaljer={
        Object {
          "arkivnokkel": "",
          "begrunnelse": "",
          "id": "",
          "kortType": "KORRIGERT_ELEKTRONISK",
          "lestDato": 2019-01-01T10:00:00.000Z,
          "meldeDato": 2019-01-01T10:00:00.000Z,
          "meldekortId": 0,
          "meldeperiode": "",
          "sporsmal": Object {
            "annetFravaer": false,
            "arbeidet": false,
            "arbeidssoker": false,
            "kurs": false,
            "meldekortDager": Array [],
            "signatur": false,
            "syk": false,
          },
        }
      }
      typeYtelse="-AAP"
    />
  </Connect(IntlProvider)>
</Provider>
`;
